include/mips/exc.h

```diff
@@ -7,7 +7,18 @@
 
 #include <mips/asm.h>
 
-#define SAVE_REG(reg, offset, base) sw reg, (EXC_##offset)(base)
+#define _ADD_CFI_NO_CFI(reg, offset)
+
+#define _ADD_CFI_(reg, offset) .cfi_rel_offset reg, offset
+
+/* Resolves to _ADD_CFI_(reg, offset) when no variadic arguments given.
+ * Resolves to _ADD_CFI_NO_CFI when NO_CFI given as one, and only one
+ * variadic argument.
+ * Passing multiple variadic arguments to this macro is considered an error.
+ */
+#define SAVE_REG(reg, offset, base, ...)                                       \
```

Comment:
Could you be more direct and rename it to `SAVE_REG_CFI`, restore original `SAVE_REG` and remove `_ADD_CFI_*` clutter?